crates/ty_python_semantic/src/types/class_base.rs

```diff
@@ -107,8 +107,19 @@ impl<'db> ClassBase<'db> {
             {
                 Self::try_from_type(db, todo_type!("GenericAlias instance"))
             }
+            Type::Intersection(inter) => {
+                let dynamic_element = inter
```

Comment:
This change looks good, but is there any reason it should be specific to a dynamic element? It seems like as long as the intersection is not disjoint from `type`, and we can find at least one element in it that is a valid base, we should be able to safely consider it that type and construct the base accordingly.

(Not saying that change needs to happen in this PR -- but the thought could maybe go into a comment, to help us out if it comes up in future.)

Reply:
That's... an interesting thought. I'm not sure it has any practical consequences right now, because the only other "atomic" types we support as class bases currently are class-literal types and dynamic types. `<dynamic type> & whatever` is handled here. `<class literal type> & whatever` should always either simplify to `<class literal type>` or `Never`, since class-literal types only have a single inhabitant.

(We probably should allow users to inherit from objects of `type[Any]`, though, due to it [actually being an intersection](https://github.com/astral-sh/ty/issues/222). And if we did that, then yes, this would have more practical consequences. I can do that in a followup. And I'll tackle this at the same time!)